moja-kuhinjica-frontend/src/service/User.service.ts

```diff
@@ -0,0 +1,40 @@
+import { axiosInstance } from 'src/config/axios'
+import { Component } from 'react'
+import { FieldValues } from 'react-hook-form'
+
+interface User {
+    id: number
+    name: string
+    surname: string
+    phoneNumber: string
+    role: string
+}
+export default class UserService extends Component {
+    public static async login(data: FieldValues): Promise<any> {
+        return await axiosInstance.post(
+            process.env.NEXT_PUBLIC_BASE_URL + '/auth/login',
+            data
+        )
+    }
+
+    public static async signIn(data: FieldValues): Promise<any> {
+        return await axiosInstance.post(
+            process.env.NEXT_PUBLIC_BASE_URL + '/client',
+            data
+        )
+    }
+
+    public static async getLoggedInUser(): Promise<any> {
+        return await axiosInstance.get(
+            process.env.NEXT_PUBLIC_BASE_URL + '/auth/profile'
+        )
+    }
+
+    public static async getUserById(
+        id: string | string[] | undefined
+    ): Promise<any> {
+        return await axiosInstance.get(
```

Comment:
When you have already created custom axios instance, you don't need `process.env.NEXT_PUBLIC_BASE_URL` , you should just create api call like this `await axiosInstance.get(`/client/${id}`)`. Because you have already added base url

Reply:
Fixed.